src/stactools/core/utils/convert.py

```diff
@@ -16,25 +17,101 @@
-    """Creates a Cloud-Optimized GeoTIFF (COG) from a GDAL-readable file.
+def assert_cog_driver_is_enabled() -> None:
+    if not utils.gdal_driver_is_enabled("COG"):
+        raise DriverRegistrationError(
+            "GDAL's COG driver is not enabled, make sure you're using GDAL >= 3.1"
+        )
 
-    Uses :py:meth:`rasterio.shutil.copy`.
+
+def cogify(
+    infile: str,
+    outfile: str,
+    band: Optional[int] = None,
+    profile: Optional[Dict[str, Any]] = None,
+) -> None:
+    """Creates a Cloud-Optimized GeoTIFF (COG) from a GDAL-readable file.
+    A band number can optionally be provided to extract a single band from a
+    multiband file. To create COGs from subdatasets, use
+    :py:meth:`stactools.core.utils.convert.cogify_subdatasets`.
 
     Args:
         infile (str): The input file.
         outfile (str): The output COG to be written.
+        band (Optional[int]): The band number in the input file to extract.
+                            If not provided, a multi-band COG will be created.
         profile (Optional[dict[str, Any]]):
             An optional profile to use on the
             output file. If not provided,
             :py:const:`stactools.core.utils.convert.DEFAULT_PROFILE` will be
             used.
     """
-    if not utils.gdal_driver_is_enabled("COG"):
-        raise DriverRegistrationError(
-            "GDAL's COG driver is not enabled, make sure you're using GDAL >= 3.1"
-        )
-    destination_profile = DEFAULT_PROFILE.copy()
+    assert_cog_driver_is_enabled()
+
+    src = rasterio.open(infile)
+    dest_profile = DEFAULT_PROFILE.copy()
+    dest_profile.update(
+        {
+            "width": src.width,
+            "height": src.height,
+            "crs": src.crs,
+            "transform": src.transform,
+        }
+    )
+
     if profile:
-        destination_profile.update(profile)
-    rasterio.shutil.copy(infile, outfile, **destination_profile)
+        dest_profile.update(profile)
+
+    # If a band number was provided, create a single-band COG
+    if band:
+        single_band = src.read(band)
+        dest_profile.update({"count": 1, "dtype": single_band.dtype})
+        with rasterio.open(outfile, "w", **dest_profile) as dest:
+            dest.write(single_band, 1)
+    # If no band numbers were provided, create a multi-band COG
+    else:
+        dest_profile.update({"count": src.count, "dtype": src.dtypes[0]})
+        rasterio.shutil.copy(infile, outfile, **dest_profile)
+
+
+def cogify_subdatasets(
+    infile: str, outdir: str, subdataset_names: Optional[List[str]] = None
+) -> Tuple[List[str], List[str]]:
+    """Creates Cloud-Optimized GeoTIFFs for all subdatasets in a multi-dataset raster file.
+    The created files will be named the same as the source file, with a ``_SUBDATASET`` suffix.
+    E.g. if the source file is named ``foo.hdf`` and the subdataset is named ``bar``, the output
+    COG will be named ``foo_bar.tif``.
```

Comment:
Probably worth adding a note about how only 2D subdatasets are supported, not 3D.